libs/wingc/src/ast.rs

```diff
@@ -652,28 +650,27 @@ pub enum InterpolatedStringPart {
 	Expr(Expr),
 }
 
-#[derive(Derivative, Default)]
-#[derivative(Debug)]
+// do not derive Default, as we want to explicitly generate IDs
+#[derive(Debug)]
 pub struct Scope {
+	/// An identifier that is unique among all scopes in the AST.
+	pub id: usize,
```

Comment:
wdyt about using a marker type for this? (Is marker type a thing? Does this have a better name?)

e.g.

```rust
/// An identifier that is unique among all scopes in the AST.
pub type ScopeId = usize
```

```suggestion
	pub id: ScopeId,
```

I feel like it helps make the documentation for it more accessible, plus if we ever want to get away from usize for it (we should) then it'll be an easy change.

Reply:
good idea!